modules/coin/transform.go

```diff
@@ -0,0 +1,65 @@
+package coin
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk"
+	"github.com/cosmos/cosmos-sdk/errors"
+)
+
+type coinTx struct {
+	Inputs  []coinInput  `json:"inputs"`
+	Outputs []coinOutput `json:"outputs"`
+}
+
+type coinInput struct {
+	Sender string `json:"sender"`
+	Coins  Coins  `json:"coins"`
+}
+
+type coinOutput struct {
+	Receiver string `json:"receiver"`
+	Coins    Coins  `json:"coins"`
+}
+
+// ExtractCoinTx makes nice json from raw tx bytes
+func ExtractCoinTx(data []byte) (interface{}, error) {
+	tx, err := sdk.LoadTx(data)
+	if err != nil {
+		return nil, err
+	}
+	txl, ok := tx.Unwrap().(sdk.TxLayer)
+	for ok {
+		tx = txl.Next()
```

Comment:
don't know the context, but a comment would be nice. are we getting the last tx in the list?